SONARJAVA-4492 Modify rule S2441: rewrite title and description (#2103)
SonarSource/rspec: rules/S2441/java/rule.adoc

== Why is this an issue?

> `HttpSession` is managed by web servers and can be serialized and stored on disk as the server manages its memory use in a process called "passivation".

Reviewer: ```suggestion
`HttpSession`s are managed by web servers and can be serialized and stored on disk as the server manages its memory use in a process called "passivation" (and later restored during "activation").
```

Even though `HttpSession` does not extend `Serializable`, you must nonetheless assume that it will be serialized.

> If non-serializable objects were stored in the session, its serialization might fail.

Reviewer: ```suggestion
If non-serializable objects are stored in the session, serialization might fail.
```
